### ISSUE TITLE
Ineffective Workflow Structure

### ISSUE THREAD

**Issue by Lainkai** [team3487/3487robotCode#3]
## Acknowledging My Mistakes
**I know that this is a really long read, but please do read all of it! There is a lot of important information about how our work flow can be streamlined, and I have put a lot of time into thinking about this. But if you are in a time crunch, I *have* made a TL;DR**

Even though I am the one that told us that we each need to have our own personal branch, I think I have found a better way to effectively solve problems. I know I told everyone to set up their accounts like this, so I am sorry.

## Explaining
Our current means of branch structure is ineffective and cumbersome. The current methods we use are not effective at updating whenever we update the master branch. 

Whenever we commit/merge to the master branch, those updates **will not** automatically push to all sub-branches, so we need to delete the current sub-branches. 
Branches, also, are not meant to be permanent features (formally), for they are meant to work on a specific issues or problems.

 I know many of you have not had any experience with Git/GitHub, and this is honestly my first time working on a group project on GitHub, but basing off of other major projects I have worked on, these changes are necessary to help streamline our workflow.

Again I am sorry about these rapid changes.

## Build Plan (in detail)
Getting used to this may take some time, but all of you are all intelligent, and I think you can pick this up really quickly. First, we need to get rid of all of our current branches, **except master**. The branches are not able to automatically update with master, and they are also just another middleman to our process. 

Next we need to get accustomed to posting on the issues board. We can use these as ways to record our progress, report issues, ask questions, and assign people to certain aspects of the project. There are labels that can also help organize these issues.  If you believe that a question is irrelevant, already answered, or just won't be fixed, there are labels for that.

After you finish your current assignment, go to the original issue that assigned you, and mark it as `done`. You can then request to merge your current build with master, and after review, it may be added to the master branch. 

If there are no conflicts, go ahead and archive/delete the old branch, since we won't need it anymore. There! You have now completed the assignment, and can now go on and search up your dankest of memes.

## TL;DR
We need to redo our current branch structure. Branches are meant for temporary use, not for permanent builds.
1. Delete all personal branches, the ones with our names.
2. Whenever we have a build plan, break down the solution into one of these 'issues', then assign ourselves to the issue.
3. Then, we break off a branch to specifically work on the issue.
4. Once the issue has been solved, you need to merge the branch to master. Make sure it has been reviewed before merging.
5. Delete the branch, after making sure all necessary changes have been made to master. Also, **make sure your build is stable before pushing to master.**
6. Close the issue, and label it as done.

### **Please make sure to see through this as soon as possible. If you have any questions, contact me.**
